Classes/DataProcessing/FilesProcessor.php

```diff
@@ -178,48 +178,38 @@ protected function processFiles(array $properties = []): ?array
+                    $delayProcessing
                 );
 
-                $targetWidth = (int)($properties['width'] ?: $file['properties']['dimensions']['width']);
-                $targetHeight = (int)($properties['height'] ?: $file['properties']['dimensions']['height']);
-
-                if (isset($this->processorConfiguration['processingConfiguration.']['autogenerate.']['retina2x']) &&
-                    (int)$this->processorConfiguration['processingConfiguration.']['autogenerate.']['retina2x'] === 1 &&
-                    ($targetWidth || $targetHeight)) {
-                    $file['urlRetina'] = $this->getFileUtility()->processFile(
-                        $fileObject,
-                        array_merge(
-                            $properties,
-                            [
-                                'width' => $targetWidth * FileUtility::RETINA_RATIO,
-                                'height' => $targetHeight * FileUtility::RETINA_RATIO,
-                            ]
-                        ),
-                        $cropVariant,
-                        (int)($this->processorConfiguration['processingConfiguration.']['delayProcessing'] ?? 0) === 1
-                    )['publicUrl'];
-                }
-
-                if (isset($this->processorConfiguration['processingConfiguration.']['autogenerate.']['lqip']) &&
-                    (int)$this->processorConfiguration['processingConfiguration.']['autogenerate.']['lqip'] === 1 &&
-                    ($targetWidth || $targetHeight)) {
-                    $file['urlLqip'] = $this->getFileUtility()->processFile(
-                        $fileObject,
-                        array_merge(
-                            $properties,
-                            [
-                                'width' => $targetWidth * FileUtility::LQIP_RATIO,
-                                'height' => $targetHeight * FileUtility::LQIP_RATIO,
-                            ]
-                        ),
-                        $cropVariant,
-                        (int)($this->processorConfiguration['processingConfiguration.']['delayProcessing'] ?? 0) === 1
-                    )['publicUrl'];
+                // 2. render autogenerate variants
+                $targetWidth = (int)($properties['width'] ?? $file['properties']['dimensions']['width']);
+                $targetHeight = (int)($properties['height'] ?? $file['properties']['dimensions']['height']);
+                if ($targetWidth || $targetHeight) {
+                    foreach ($this->processorConfiguration['processingConfiguration.']['autogenerate.'] as $formatKey => $formatConf) {
+                        $formatKey = rtrim($formatKey, '.');
+                        $factor = (float)($formatConf['factor'] ?? 1.0);
+
+                        $file['publicUrl_' . $formatKey] = $this->getFileUtility()->processFile(
```

Comment:
use camelCase key in response, also you introduced breaking change.

Reply:
Yes, it was not documented, but  we do not have closed source software. 
If someone is using this feature, you will break this feature for them, also BE dev need to adjust typoscript configuration and FE also need to adjust for new response. Ability to configure more options via typoscript it's nice, but we should have fallbacks for now (until version `5.0`)

Reply:
Seems to be working and rendering the same output as before...